ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java

```diff
@@ -397,22 +469,13 @@ private String getValidWriteIdList(String dbName, String tblName, String validTx
     return openTxns;
   }
 
-  String getValidTxnListForReplDump(Hive hiveDb) throws HiveException {
-    // Key design point for REPL DUMP is to not have any txns older than current txn in which dump runs.
-    // This is needed to ensure that Repl dump doesn't copy any data files written by any open txns
-    // mainly for streaming ingest case where one delta file shall have data from committed/aborted/open txns.
-    // It may also have data inconsistency if the on-going txns doesn't have corresponding open/write
-    // events captured which means, catch-up incremental phase won't be able to replicate those txns.
-    // So, the logic is to wait for configured amount of time to see if all open txns < current txn is
-    // getting aborted/committed. If not, then we forcefully abort those txns just like AcidHouseKeeperService.
-    ValidTxnList validTxnList = getTxnMgr().getValidTxns();
-    long timeoutInMs = HiveConf.getTimeVar(conf,
-            HiveConf.ConfVars.REPL_BOOTSTRAP_DUMP_OPEN_TXN_TIMEOUT, TimeUnit.MILLISECONDS);
-    long waitUntilTime = System.currentTimeMillis() + timeoutInMs;
+  ValidTxnList getValidTxnListForReplDump(Hive hiveDb, ValidTxnList validTxnList,
```

Comment:
As discussed earlier, validTxnList need not be obtained from caller. Also, shall return String itself.

Reply:
See my explanation above about validTxnList being passed.

Earlier the transaction snapshot was being obtained within getValidTxnListFoReplDump(), and thus was fine to return a String representation. Now that we are passing snapshot to this function, it's easier to return it as is if required. Furthermore the caller then use the snapshot object as is or convert to a string as required - bit of flexibility to the caller in that case. That's why I have changed the return type to snapshot instead of string, which needs to be parsed back to get snapshot object if that's required. Since there are only two callers it's an easy change to make now that later in case the number of callers increases.

Reply:
As discussed, getting ValidTxnList inside getValidTxnListForReplDump() doesn't make any difference as it doesn't include the txns opened after currentTxn. Refer to TxnCommonUtils.createValidReadTxnList() for this.
So, please revert back the changes as proposed in my comment.

Reply:
Done.